src/m3ninx/index/segment/fst/writer.go

```diff
@@ -67,20 +77,37 @@ type WriterOptions struct {
 	// amount (e.g. 2x). You can disable this to speed up high fixed cost
 	// lookups to during building of the FST however.
 	DisableRegistry bool
+
+	// ForTestsOnly_Version allows users to override behaviour of the
+	// writer. NB: this is intended to be used only during tests.
+	ForTestsOnly_Version *Version
 }
 
 // NewWriter returns a new writer.
-func NewWriter(opts WriterOptions) Writer {
-	return &writer{
-		intEncoder:      encoding.NewEncoder(defaultInitialIntEncoderSize),
-		postingsEncoder: pilosa.NewEncoder(),
-		fstWriter:       newFSTWriter(opts),
-		docDataWriter:   docs.NewDataWriter(nil),
-		docIndexWriter:  docs.NewIndexWriter(nil),
-		postingsOffsets: make([]uint64, 0, defaultInitialPostingsOffsetsSize),
-		fstTermsOffsets: make([]uint64, 0, defaultInitialFSTTermsOffsetsSize),
-		docOffsets:      make([]docOffset, 0, defaultInitialDocOffsetsSize),
+func NewWriter(opts WriterOptions) (Writer, error) {
+	v := CurrentVersion
+	if opts.ForTestsOnly_Version != nil {
+		v = *opts.ForTestsOnly_Version
+	}
+	if err := v.Supported(); err != nil {
+		return nil, err
 	}
+
+	return &writer{
+		version:             v,
+		intEncoder:          encoding.NewEncoder(defaultInitialIntEncoderSize),
+		postingsEncoder:     pilosa.NewEncoder(),
+		fstWriter:           newFSTWriter(opts),
+		docDataWriter:       docs.NewDataWriter(nil),
+		docIndexWriter:      docs.NewIndexWriter(nil),
+		docOffsets:          make([]docOffset, 0, defaultInitialDocOffsetsSize),
+		fstTermsOffsets:     make([]uint64, 0, defaultInitialFSTTermsOffsetsSize),
+		termPostingsOffsets: make([]uint64, 0, defaultInitialPostingsOffsetsSize),
+
+		fieldPostingsOffsets: make([]uint64, 0, defaultInitialPostingsOffsetsSize),
+		fieldsPostingsList:   roaring.NewPostingsList(),
```

Comment:
I think you need to actually using a roaring postings list created from a bitmap that has container pooling turned on (as per the multi segment iterator stuff).

Otherwise it just throws out it's containers every time you call `Reset()`.

Sorry, do you mind updating to do that?

Reply:
sure thing. Thanks for the catch. Fixed by creating a pilosa bitmap backed by a pool and wrapping into a postings list (need to cleanup the leakiness of postings list abstraction in a followup)